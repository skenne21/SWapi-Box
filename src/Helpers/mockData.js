const mockMovie = () => {
  return {
    title: "The Empire Strikes Back",
    episodeId: 5,
    opening_crawl: 'It is a dark time for the...',
    director: "Irvin Kershner", 
    release_date: '1980-05-17'
  };
};

const mockApiPeople = () => {
  return [
    {
      name: "Luke Skywalker", 
      height: "172", 
      mass: "77", 
      hairColor: 
      "blond", 
      skinColor: 
      "fair", 
      homeworld:"https://swapi.co/api/planets/1/",
      species:["https://swapi.co/api/species/1/"]
    }, {
      name: "C-3PO",
      height: "167", 
      mass: "75", 
      hairColor: "n/a", 
      skinColor: "gold",
      homeworld:"https://swapi.co/api/planets/1/",
      species:["https://swapi.co/api/species/1/"]
    }];
};

const mockHomelandData = () => {
  return [
    {
      name: "Luke Skywalker",
      homeworld:"https://swapi.co/api/planets/1/",
      species:["https://swapi.co/api/species/1/"],
      population: undefined
    }, {
      name: "C-3PO",
      homeworld:"https://swapi.co/api/planets/1/",
      species:["https://swapi.co/api/species/1/"],
      population: undefined
    } 
  ];
};

const mockSpecies = () => {
  return [
    {
      name: "Luke Skywalker",
      homeworld: 'Somewhere',
      species: "https://swapi.co/api/species/1/",
      population: 30000
    }, {
      name: "C-3P0",
      homeworld: 'Somewhere',
      species: "https://swapi.co/api/species/1/",
      population: 30000
    }
  ];
};

const mockCleanPeople = () => {
  return[
    {
      name: "Luke Skywalker",
      homeworld: 'Somewhere',
      species: 'human',
      population: 30000
    }, {
      name: "C-3P0",
      homeworld: 'Somewhere',
      species: 'human',
      population: 30000
    }
  ]
}

const mockApiPlanets = () => {
  return [{
    climate:"temperate",
    name: "Alderaan",
    population:"2000000000",
    residents:["https://swapi.co/api/people/5/", "https://swapi.co/api/people/68/", "https://swapi.co/api/people/81/"],
    terrain:"grasslands, mountains"
  }];
};



export default {
  mockMovie, 
  mockApiPeople, 
  mockHomelandData, 
  mockSpecies,
  mockApiPlanets, 
  mockCleanPeople
};