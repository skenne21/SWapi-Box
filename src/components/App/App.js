import React, { Component } from 'react';
import Nav from '../Nav/Nav.js';
import Main from '../Main/Main.js';
import Header from '../Header/Header.js';
import swapiData from '../../helper.js'


import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      film: {},
      favorites: [],
      cards: []
    }
  }

  getMovie = () => {
    swapiData.fetchMovie()
      .then(apiData => {
        const movie = swapiData.cleanMovie(apiData);
        this.setState({film: movie})
      })
      .catch(error => console.log(error));
  }

  setCardsState = (apiData) => {
    this.setState({cards: apiData})
  }

  getCards = (userInput) => {
    if (userInput === 'People') {
      const people = swapiData.fetchPeople()
        .then(apiData => this.setCardsState(apiData))
    }
    if (userInput === 'Planets') {
      const planets = swapiData.fetchPlanets()
        .then(apiData => this.setCardsState(apiData))
    }
    if (userInput === 'Vehicles') {
      const vehicles = swapiData.fetchVehicles()
        .then( apiData => this.setCardsState(apiData))
    }
     
  }
  
  componentDidMount() {
    // this.getMovie();
  }

  render() {
    const { favorites, film, cards } = this.state; 
    return (
      <div className="App">
        <Header 
          favorites={favorites}/>
        <Nav
          controlFunc={this.getCards}/>
        <Main 
          film={film}
          cards={cards}/>
      </div>
    );
  }
}

export default App;


