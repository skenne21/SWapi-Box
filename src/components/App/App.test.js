import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import App from './App';
import mockData from '../../Helpers/mockData.js';

describe("App shallow", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />, {disableLifecycleMethods: true});
  });

  it("Should match the snapShot" , () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should a default state empty object for films', () => {
    expect(wrapper.state('film')).toEqual({});
  });

  it('Should a default state empty array for favorites', () => {
    expect(wrapper.state('favorites')).toEqual([]);
  });

  it('Should a default state empty array for cards', () => {
    expect(wrapper.state('cards')).toEqual([]);
  });

  it('Should a default state error of false', () => {
    expect(wrapper.state('error')).toEqual(false);
  });

  it('Should a default state of isActive, to a empty string', () => {
    expect(wrapper.state('isActive')).toEqual('');
  });

  it('Should a default state of targetFavorites to false', () => {
    expect(wrapper.state('targetFavorites')).toEqual(false);
  });

  it('Should set the sate of isActive with the user input', async ()  => {
    expect(wrapper.state('isActive')).toEqual('');
    const getCards =await wrapper.instance().getCards('People', {});
    expect(wrapper.state('isActive')).toEqual('People');
    const getCardsAgain = await wrapper.instance().getCards('Vehicles', {});
    expect(wrapper.state('isActive')).toEqual('Vehicles');
  })

  it('Should call removeFavorites if card is in array', () => {
    const retrieved = {
      name: "C-3P0",
      homeworld: 'Somewhere',
      species: 'human',
      population: 30000
    };
    const spy = jest.spyOn(wrapper.instance(), 'removeFavorites');
    wrapper.setState({favorites: [retrieved]});
    wrapper.instance().toggleFavorites('people', retrieved);
    expect(spy).toHaveBeenCalledWith(retrieved);
  });

  it('Should call addFavorties when card is in array', () => {
    const retrieved = {
      name: "Luke Skywalker",
      homeworld: 'Somewhere',
      species: 'human',
      population: 30000
    };
    wrapper.setState({favorites: []});
    const spy = jest.spyOn(wrapper.instance(), 'addFavorites');
    wrapper.instance().toggleFavorites('people', retrieved);
    expect(spy).toHaveBeenCalledWith(retrieved);
  });

  it('Should remove the card that was passed from the favorites', () => {
    const retrieved = {
      name: "Luke Skywalker",
      homeworld: 'Somewhere',
      species: 'human',
      population: 30000
    };
    expect(wrapper.state('favorites')).toEqual([]);
    wrapper.instance().addFavorites(retrieved);
    expect(wrapper.state('favorites')).toEqual([retrieved]);
  });

  it('Should set the state of targetFavorites to true', () => {
    expect(wrapper.state('targetFavorites')).toEqual(false);
    wrapper.instance().showFavorites();
    expect(wrapper.state('targetFavorites')).toEqual(true);
  });

  it('Should set the state of cards to the favorites array', () => {
    const cards = [{
      name: "C-3P0",
      homeworld: 'Somewhere',
      species: 'human',
      population: 30000
    }];
    const favorites = [{
      name: "Luke Skywalker",
      homeworld: 'Somewhere',
      species: 'human',
      population: 30000
    }];
    wrapper.setState({cards: cards});
    expect(wrapper.state('cards')).toEqual(cards);
    wrapper.setState({favorites: favorites});
    wrapper.instance().showFavorites();
    expect(wrapper.state('cards')).toEqual(favorites);
  });
  
})



