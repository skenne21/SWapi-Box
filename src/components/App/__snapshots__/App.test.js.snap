// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App shallow Should match the snapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          favorites={Array []}
          showFavorites={[Function]}
/>,
        <Nav
          controlFunc={[Function]}
          isActive=""
/>,
        <Main
          cards={Array []}
          favorites={Array []}
          film={Object {}}
          showFavorites={[Function]}
          targetFavorites={false}
          toggleFavorites={[Function]}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "favorites": Array [],
          "showFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "controlFunc": [Function],
          "isActive": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "cards": Array [],
          "favorites": Array [],
          "film": Object {},
          "showFavorites": [Function],
          "targetFavorites": false,
          "toggleFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            favorites={Array []}
            showFavorites={[Function]}
/>,
          <Nav
            controlFunc={[Function]}
            isActive=""
/>,
          <Main
            cards={Array []}
            favorites={Array []}
            film={Object {}}
            showFavorites={[Function]}
            targetFavorites={false}
            toggleFavorites={[Function]}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "favorites": Array [],
            "showFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "controlFunc": [Function],
            "isActive": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "cards": Array [],
            "favorites": Array [],
            "film": Object {},
            "showFavorites": [Function],
            "targetFavorites": false,
            "toggleFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
